Rewrite docs (#118)
kubeshop/kusk-gateway: docs/installation.md

---
**NOTE**

This documents describes the installation of the Kusk Gateway and its load balancing component Envoy Fleet to the generic Kubernetes cluster.

If you're looking for the quick way to try Kusk Gateway in a locally setup Kubernetes cluster, please see [Local Installation with Minikube](local-installation.md).

---

# Table of contents
- [Prerequsities](#prerequsities)
  - [Cluster requirements](#cluster-requirements)
  - [Install requirements](#install-requirements)
- [Installation](#installation)
- [Uninstallation](#uninstallation)

During the setup we'll deploy Kusk Gateway Custom Resources Definitions, Kusk Gateway Manager and Envoy Fleet with Helm.

For the architectural overview of the components please check the [Architecture](arch.md) page.

## Prerequsities

### Cluster requirements

- Kubernetes v1.16+

- Kubernetes cluster administration rights are required - we install CRDs, service account with ClusterRoles and RoleBindings.

- If you don't have Jetstack Cert-Manager installed in your cluster, then please follow the official [instructions](https://cert-manager.io/docs/installation/) to setup it. We use Cert-Manager for the webhooks configuration.

> - If you have the managed cluster (GCP, EKS, etc) then you can skip reading this paragraph.

Reviewer: ```suggestion
- If you have the managed cluster (GCP, EKS, etc) then you can skip to the next section
```
